### PR TITLE
feat: store initial timeline in env fixture

### PR DESCRIPTION
**Pull request by koivunej**
really minor change, but will allow more use in future for the default tenants.

separated from #3809, and separate from rewriting all the single tenant tests to use these values instead..

### CODE REVIEW

**Review by LizardWizzard — APPROVED** [review]
In general I see that it may be benefitial to break down `env` to separate fixtures. We can have `tenant` fixture as a parameter for tests that need it

**Review by SomeoneToIgnore — APPROVED** [review]

File: test_runner/fixtures/neon_fixtures.py [hunk 1]
```diff
@@ -904,6 +905,7 @@ def __init__(self, config: NeonEnvBuilder):
         # generate initial tenant ID here instead of letting 'neon init' generate it,
         # so that we don't need to dig it out of the config file afterwards.
         self.initial_tenant = config.initial_tenant
+        self.initial_timeline: None | TimelineId = None
```
Comment (bayandin) [inline test_runner/fixtures/neon_fixtures.py:908]: I think we usually use it with `Optional`:

```suggestion
        self.initial_timeline: Optional[TimelineId] = None
```


Note: I think we can enforce the style of this with `ruff` (https://beta.ruff.rs/docs/rules/#pyupgrade-up), but it'll change a lot of code, so I'd do it in a separate PR

Comment (koivunej) [reply test_runner/fixtures/neon_fixtures.py:908]: Aah yes, apologies on this. So obvious when I see your suggestion, but honestly didn't even remember python types had Optional. Which ... I think is surprising for others as well, since the gradual typing fully supports this `None | something_else`..

I'll commit this in anyways, since I guess it's more typed-pythonic.

**Comment by koivunej** [comment]
In case anyone goes to look how is this used in #3809, don't waste your time, it is used, but very slightly and indirectly. I was initially thinking of using it more as one of the tenants I create, but it wasn't necessary because I added `testing_api!(PUT /v1/:tenant_id/broken)` to break a tenant which is also useful in our test.

I don't think we are in a hurry to merge this, but it should go in before #3809. I'll leave it open for now to raise awareness.

**Comment by koivunej** [comment]
(it's FAILED test_runner/regress/test_compatibility.py::test_forward_compatibility@compatibility, will retry)
